dace/codegen/compiled_sdfg.py

```diff
@@ -230,7 +230,7 @@ def _try_parse_state_struct(self) -> Optional[Type[ctypes.Structure]]:
             field_str = field_str.strip()
 
             match_name = re.match(
-                r'(?:const)?\s*(.*)(?:\s+\*\s*|\s*\*\s+)([a-zA-Z_][a-zA-Z_0-9]*)$',
+                r'(?:const)?\s*(.*)(?:\s+\*\s*|\s*\*\s+\_\_restrict\_\_\s+)([a-zA-Z_][a-zA-Z_0-9]*)$',
```

Comment:
The declaration of persistent memory is now created by `Data.as_arg` (see framecode.py, line 484), which adds the `__restrict__` keyword (and probably makes sense since, similarly to program arguments, we don't want it to alias). The change above makes sure that we can properly parse structs (in this case program states) that contain `__restrict__` pointers (see tests/parse_state_struct_test.py).